plugins/imagebase/plugin.js

```diff
@@ -832,6 +842,44 @@
 		this.bar.setStyle( 'width', percentage + '%' );
 	};
 
+	ProgressBar.prototype.bindEditor = function( editor ) {
+		var undoListeners = [],
+			parent = this.wrapper.getParent();
+
+		// Hide progressbar during snapshot creation so it does not trigger saving new snapshot
+		// (as content will be changed due to progressbar progress).
+		undoListeners.push(
+			editor.on( 'beforeUndoImage', function() {
+				if ( this.wrapper.getParent() ) {
+					if ( !parent ) {
+						parent = this.wrapper.getParent();
+					}
+					// Detach wrapper only if it is attached.
+					this.wrapper.remove();
+				}
+			}, this )
+		);
+
+		undoListeners.push(
+			editor.on( 'afterUndoImage', function() {
+				if ( !this.wrapper.getParent() && parent ) {
+					// Attach wrapper only if it is detached.
+					parent.append( this.wrapper );
+				}
+			}, this )
+		);
+
+		var remove = this.remove;
+		this.remove = function() {
+			// Remove undo listeners when progress bar is removed.
+			CKEDITOR.tools.array.filter( undoListeners, function( listener ) {
```

Comment:
Why `filter` is used here? It does not modify original array, so it's left untouched, with all listeners inside. `CKEDITOR.tools.array.forEach` + `undoListener.length = 0` seems like a better alternative.

Reply:
Looks like some leftover after poor copy/paste 😭